beacon-chain/core/helpers/randao.go

```diff
@@ -0,0 +1,82 @@
+package helpers
+
+import (
+	"fmt"
+
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"github.com/prysmaticlabs/prysm/shared/hashutil"
+)
+
+// GenerateSeed generates the randao seed of a given epoch.
+//
+// Spec pseudocode definition:
+//   def generate_seed(state: BeaconState,
+//                  epoch: EpochNumber) -> Bytes32:
+//    """
+//    Generate a seed for the given ``epoch``.
+//    """
+//    return hash(
+//        get_randao_mix(state, epoch - SEED_LOOKAHEAD) +
+//        get_active_index_root(state, epoch)
+//    )
+func GenerateSeed(state *pb.BeaconState, wantedEpoch uint64) ([32]byte, error) {
+	if wantedEpoch > config.SeedLookahead {
```

Comment:
Any reason this chunk isnt in the spec ?
```
	if wantedEpoch > config.SeedLookahead {
		wantedEpoch -= config.SeedLookahead
	}
```

Reply:
to prevent under flow, spec aka python code don't explicitly specify under flow conditions, for these seeds we are always looking `config.SeedLookahead` behind